core/src/utilities.zig

```diff
@@ -188,6 +189,60 @@ pub fn Slice_Vector(comptime Slice: type) type {
     };
 }
 
+pub fn max_enum_tag(T: type) @typeInfo(T).Enum.tag_type {
+    var max_tag: @typeInfo(T).Enum.tag_type = 0;
+    for (@typeInfo(T).Enum.fields) |field| {
+        if (field.value > max_tag) {
+            max_tag = field.value;
+        }
+    }
+    return max_tag;
+}
+
+pub fn GenerateInterruptEnum() type {
+    var fields: [microzig.chip.INTERRUPTS.len]std.builtin.Type.EnumField = undefined;
```

Comment:
Similar to above, maybe we bail out if INTERRUPTS is empty. But maybe this works fine with 0 fields?